### PR TITLE
chore: use pnpm 7

### PR DESCRIPTION
**Pull request by bluwy**
Use pnpm 7. Drop node12

### CODE REVIEW

**Comment by bluwy** [comment]
Findings:

1. `corepack enable` can't be used in CI as it can only be called after `setup-node`, but `setup-node` calls the `pnpm` binary while initializing. There's a feature request to auto enable corepack within `setup-node`

File: packages/vite-plugin-qrcode/package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "test": "tsm node_modules/uvu/bin.js tests"
   },
   "engines": {
-    "node": "^12.20 || ^14.13.1 || >= 16"
+    "node": ">=14"
```
Comment (dominikg) [inline packages/vite-plugin-qrcode/package.json:27]: this allows node 15 and 17, both of them are EOL

```suggestion
    "node": "^14.13.1 || ^16 || >=18"
```

File: packages/vite-plugin-qrcode/tsup.config.js [hunk 2]
```diff
@@ -3,5 +3,6 @@ export const tsup = {
 	entryPoints: ['src/index.ts'],
 	format: ['esm', 'cjs'],
 	splitting: false,
-	clean: true
+	clean: true,
+	target: 'node14'
```
Comment (dominikg) [inline packages/vite-plugin-qrcode/tsup.config.js:7]: use min node version as target
```suggestion
	target: 'node14.13.1'
```

**Comment by bluwy** [comment]
Update node version to `^14.13.1 || ^16.0.0 || >=18` 👍

**Pull request merged**